tech-specs/src/explore/design/dkg.md

```diff
@@ -1 +1,178 @@
+consider the instances ranked in terms of their validators as was also used
+for partitioning the weight shares (described above).
+
+## Storage considerations
+
+The storage of a node must be adapted to accommodate the DKG protocol. Each
+instance of a DKG protocol from Ferveo is a state machine that stores
+much of the necessary data from the protocol instance. The following data
+needs to be stored by each node
+ - The partition of the weight shares among the current validator set
+ - A session keypair for the current DKG session. Used for encrypting
+   PVSS transcripts and signing DKG protocol messages.
+ - A list of session ids from the last completed DKG instance up until the
+   latest started (we may prune ids for failed instances however) 
+ - The DKG state machine associated to each of these session ids.
+
+##  The DKG state machine
+
+We first describe each new message type needed for the protocol.
+- `Deal`: This message contains the PVSS transcript from a validator
+  acting as a dealer
+- `Aggregate`: Add the combined PVSS transcripts as well as each transcript
+  included in the aggregation in the header of a proposed block
+- `Complain`: A message stating that a given validator's PVSS transcript or
+  a block proposer's aggregation failed verification. 
+
+In Ferveo, the DKG protocol is modelled as a state machine. This allows a
+validator to initialize an instance and drive it forward by reacting to
+each of the above message in the appropriate way. All necessary state
+is persisted by this state machine. We describe the  appropriate action for
+each of the above transactions.
+
+ - `Deal`: These need to be verified. Each validator adds every deal message
+   it receives to the state machine and when the threshold is reached,
+   the state machine automatically performs the aggregation. The subsequent
+   block proposer checks if the aggregation was performed. If so, it adds 
+   the result to the block it is proposing. In either case, it also adds the
+   PVSS transcripts to the block header. If aggregation was successful, these
+   act as verifications. Otherwise, the next block proposer will need the 
+   information to try and complete the  protocol in the next block. For any
+   PVSS transcript that fails verification, a `Complain` message is also 
+   included in the header.
+ - `Aggregate`: This simply needs to be verified and the resulting key added
+   to the DKG instance along with the input PVSS transcripts. If validation
+   of the aggregated key fails, a `Complain` should be issued.
+
+While every validator must send and react to `Deal` messages, `Aggregate` 
+must be made by the current block proposer if their current DKG state machine 
+is in the  appropriate state and/or the appropriate conditions are met. Thus 
+block proposers must check the state machines of their active DKG instances.
+
+## ABCI++
+
+The creation of the public key and the shared secrets requires ABCI++ as we
```

Comment:
cite the ABCI++ spec (link)

Reply:
I think it is actually more explanatory to cite the RFC.